app/Place/PlaceImportServiceProvider.php

```diff
@@ -35,30 +36,28 @@ function (Application $app) {
             }
         );
 
-        $app['place_importer'] = $app->share(
-            function (Application $app) {
-                $placeImporter = new PlaceDocumentImporter(
-                    $app['place_repository'],
-                    $app['place_denormalizer'],
-                    $app['import_image_collection_factory'],
-                    $app['imports_command_bus'],
-                    $app['should_auto_approve_new_offer'],
-                    $app['labels.labels_locked_for_import_repository']
-                );
-
-                $termPreProcessor = new TermPreProcessingDocumentImporter(
-                    $placeImporter,
-                    new PlaceCategoryResolver()
-                );
-
-                $labelPreProcessor = new LabelPreProcessingDocumentImporter(
-                    $termPreProcessor,
-                    $app[LabelServiceProvider::JSON_READ_REPOSITORY],
-                    $app[LabelServiceProvider::RELATIONS_READ_REPOSITORY]
-                );
-
-                return $labelPreProcessor;
-            }
+        $app[ImportPlaceRequestHandler::class] = $app->share(
+            fn (Application $application) => new ImportPlaceRequestHandler(
+                $app['place_repository'],
+                $app['uuid_generator'],
+                $app['place_denormalizer'],
+                new CombinedRequestBodyParser(
+                    new ImportLabelVisibilityRequestBodyParser(
+                        $app[LabelServiceProvider::JSON_READ_REPOSITORY],
+                        $app[LabelServiceProvider::RELATIONS_READ_REPOSITORY]
+                    ),
+                    new ImportTermRequestBodyParser(
+                        new PlaceCategoryResolver()
+                    )
+                ),
+                $app['place_iri_generator'],
+                $app['imports_command_bus'],
+                $app['import_image_collection_factory'],
+                $app['labels.labels_locked_for_import_repository'],
+                $app['should_auto_approve_new_offer'],
+                $app['auth.api_key_reader'],
+                $app['auth.consumer_repository']
+            )
```

Comment:
Maybe move this request handler definition to `PlaceControllerProvider` with the other place request handlers, and remove this service provider. (Like Organizers)

Reply:
Indeed a cleaner approach.